x/superfluid/keeper/stake.go

```diff
@@ -243,7 +245,11 @@ func (k Keeper) SuperfluidDelegate(ctx sdk.Context, sender string, lockID uint64
 		return types.ErrOsmoEquivalentZeroNotAllowed
 	}
 
-	return k.mintOsmoTokensAndDelegate(ctx, amount, acc)
+	newShares, err := k.mintOsmoTokensAndDelegate(ctx, amount, acc)
+
+	events.EmitDelegationEvent(ctx, valAddr, newShares, amount)
```

Comment:
if there's an error here, do we still want to emit events?

Reply:
We do not

Reply:
Good catch, Thanks 👍